impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -329,8 +392,55 @@ public ClientBuilder setOrgUrl(String baseUrl) {
         return this;
     }
 
+    @Override
+    public ClientBuilder setAuthorizationMode(String authorizationMode) {
+        // if 'authorizationMode' is ever supplied, it should only be "PrivateKey" (OAUTH2 flow);
+        // else do not set it as it defaults to SSWS flow.
+        if (authorizationMode != null) {
+            if (authorizationMode.equals("PrivateKey")) {
+                this.clientConfig.setAuthorizationMode(authorizationMode);
+            }
+            else {
+                throw new IllegalArgumentException("Invalid authorizationMode");
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setScopes(List<String> scopes) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(scopes, "Missing scopes");
+            Assert.isTrue(scopes.size() > 0, "Empty scopes list");
+            this.clientConfig.setScopes(scopes);
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setPrivateKey(String privateKey) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(privateKey, "Missing privateKey");
+            this.clientConfig.setPrivateKey(privateKey);
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setClientId(String clientId) {
+        ConfigurationValidator.assertClientId(clientId);
+        this.clientConfig.setClientId(clientId);
+        return this;
+    }
+
+    boolean isOAuth2Flow() {
+        String authorizationMode = this.getClientConfiguration().getAuthorizationMode();
+        return authorizationMode != null && authorizationMode.equals("PrivateKey");
```

Comment:
See above comment about not needing `authorizationMode`

Reply:
@bdemers In one of my comments above, I laid out the reason why I'm using this attribute. We can debate about this I guess but we might need to rope in the product ppl.

Reply:
I'm not going to block on this one as it sounds like you have talked about it with the team.


My personal pref would be to detect the configuration as much as possible, currently, you need two properties, `orgUrl` and `token`, in those cases we could assume `SSWS`

Similar with the private key, we could assume if `orgUrl`, `privateKey`, and `scopes` are set we could assume `PrivateKey`.

This could also be done as a future optimization (and would be backward compatible)

Reply:
@bdemers totally agree on the potential optimizations that we could do with the present configuration. We could use this as an opportunity to clean that up but considering that this PR is on hold for long, we could consider jotting down the list of optimizations and attack them via new tasks? That's my humble suggestion. Thoughts?